internal/testutil/server.go

```diff
@@ -52,16 +52,24 @@ type Server struct {
 	Gsrv *grpc.Server
 }
 
-// NewServer creates a new Server. The Server will be listening for gRPC connections
-// at the address named by the Addr field, without TLS.
+// NewServer creates a new Server on localhost. The Server will be listening for
+// gRPC connections at the address named by the Addr field, without TLS.
 func NewServer(opts ...grpc.ServerOption) (*Server, error) {
 	return NewServerWithPort(0, opts...)
 }
 
-// NewServerWithPort creates a new Server at a specific port. The Server will be listening
-// for gRPC connections at the address named by the Addr field, without TLS.
+// NewServerWithPort creates a new Server on localhost at a specific port. The
+// Server will be listening for gRPC connections at the address named by the
+// Addr field, without TLS.
 func NewServerWithPort(port int, opts ...grpc.ServerOption) (*Server, error) {
-	l, err := net.Listen("tcp", fmt.Sprintf("localhost:%d", port))
+	return NewServerWithHost(fmt.Sprintf("localhost:%d", port))
+}
+
+// NewServerWithHost creates a new Server with a specific host and port. The
+// Server will be listening for gRPC connections at the address named by the
+// Addr field, without TLS.
+func NewServerWithHost(host string, opts ...grpc.ServerOption) (*Server, error) {
```

Comment:
Currently this function only takes in the address, and not a host. I recommend either renaming this to be `NewServerWithAddress` if you want to keep it as is, or have a separate option to specify the port if you want to keep the function name.

Same applies for the functions in `fake.go`